dbgap_extract.py

```diff
@@ -36,55 +133,55 @@ def scrape(studies_to_scrape, output_filename):
         "study_accession",
         "study_accession_with_consent",
         "study_with_consent",
+        "datastage_subject_id",
     ]
 
+    if os.path.exists(output_filename):
+        os.remove(output_filename)
+
     write_list_of_rows_to_tsv([field_names], output_filename)
 
-    for study_accession in studies_to_scrape:
+    # Use a queue to make it appropriate to modify the list during iteration
```

Comment:
why can't you just iterate over the list? why do you need to modify it? would be simpler to just use the existing for loop

Reply:
hmm i could iterate over the list, but if during the iteration over the study accessions it is discovered that an older version needs to be retrieved, we need to redo the scraping code for that older version. one way to do that would be to append that older version to the list of studies. but that would involve modifying the list we're iterating over. which is why i used a queue

Reply:
ahhh, I see now, I was missing the part where you add in the previous version